ui-performance/index.md

```diff
@@ -345,17 +372,29 @@ When you encounter calls to `saveLayer`, ask yourself these questions:
 
 * Does the app need this effect?
 * Can any of these calls be eliminated?
+* Can I apply the same effect to an individual element instead of a group?
 
 #### Checking for non-cached images
 
-Image I/O, such as passing images from host memory to device memory,
-can be expensive. Where possible, you want to cache images.
+One of the most expensive operations, from a resource perspective, is
+rendering a texture using an image file. First, the compressed image is
+fetched from persistent storage (disk, for example).
+The image is decompressed into device memory (memory on the GPU). The
+decompressed image is then passed to the host memory (RAM on the phone).
+
+In other words, image I/O can be expensive.
+The cache provides snapshots of complex hierarchies so they are easier to
+render in subsequent frames.
+_You want to cache images only where absolutely necessary._
```

Comment:
Why? Because raster cache entries are expensive to construct and take up loads of GPU memory.

Reply:
Done